ir/instr.cpp

```diff
@@ -1379,7 +1380,18 @@ StateValue Malloc::toSMT(State &s) const {
   auto &[sz, np] = s[*size];
   s.addUB(np);
   // TODO: malloc's alignment is implementation defined.
-  return { s.getMemory().alloc(sz, 8, true), true };
+  unsigned new_bid;
+  auto p = s.getMemory().alloc(sz, 8, true, &new_bid);
+
+  if (isNonNull)
+    return { move(p), true };
+
+  auto nullp = Pointer::mkNullPointer(s.getMemory());
+  stringstream ss;
+  ss << "malloc_" << new_bid << "_isnull_" << (s.isSource() ? "src" : "tgt");
+  auto flag = expr::mkBoolVar(ss.str().c_str());
```

Comment:
Please add a API for this: `smt::expr::mkFreshVar(const char *prefix, const expr &type)`.
This should call Z3_mk_fresh_var().

Reply:
Probably there are other places where we should have used this already

Reply:
Updated, let freeze use it too